test/single_column_ARS_test.jl

```diff
@@ -0,0 +1,206 @@
+    P, dP = ss.P, ss.dP
+    Δz = ss.Δz
+
+    # N cells
+    N = div(length(u) - 1 , 3)
+   
+    ρ, ρθ, w = u[1:N], u[N+1:2N], u[2N+1:3N+1]
+    
+    # interpolate ρ and ρθ to cell face values
+    ρh =  [ρ[1] ; (ρ[1:N-1] + ρ[2:N])/2.0;  ρ[N]]
+    ρθh = [ρθ[1]; (ρθ[1:N-1] + ρθ[2:N])/2.0; ρθ[N]]
+    
+    # pressure and its derivative with respect to ρθ at cell center
+    pc = P.(ρθ)
+    dpc_dρθ = dP.(ρθ)
+
+    # half cell size
+    Δzh = [NaN64; (Δz[1:N-1] + Δz[2:N])/2.0; NaN64]
+    
+    # P = ([zeros(N,N)     zeros(N,N)      ∂ρ/∂w ;
+    #       zeros(N,N)     zeros(N,N)      ∂ρθ/∂w  
+    #       ∂w/∂ρ          ∂w/∂ρθ      zeros(N+1,N+1)])
+
+    for i = 1:N
+        J[i,   i+2N]     =   ρh[i]/Δz[i]
+        J[i,   i+2N+1]   =  -ρh[i+1]/Δz[i]
+    end
+
+    for i = 1:N
+        J[i+N, i+2N]     = ρθh[i]/Δz[i]
+        J[i+N, i+2N+1]   = -ρθh[i+1]/Δz[i]
+    end
+
+    for i = 2:N #(dw_i+1/2)
+        J[i+2N, (i-1)]     =  (pc[i] - pc[i-1])/Δzh[i]/(2*ρh[i]^2)
+        J[i+2N, (i-1)+1]   =  (pc[i] - pc[i-1])/Δzh[i]/(2*ρh[i]^2)
+
+        J[i+2N, (i-1)+N]         =  dpc_dρθ[i-1]./(ρh[i]*Δzh[i])
+        J[i+2N, (i-1)+1+N]       = -dpc_dρθ[i]./(ρh[i]*Δzh[i])
+    end
+end
+
+function single_column_Wfact!(J,u,ss,γ,t)
+    J .= 0.0
+    jacobian!(J,u,ss,t)
+    J .*=γ
+    N = ss.N
+    for i = 1:3N+1
+        J[i,i] -= 1.0
+    end
+    return J
+end
+
+function linsolve_direct(::Type{Val{:init}}, f, u0; kwargs...)
+    function _linsolve!(x, A, b, update_matrix = false; kwargs...)
+        x .= A \ b
+    end
+end
+
+@testset "IMEX ARS343 on a single column" begin
+
+    N = 30
+    L = 30e3
+    N_iter = 1
+    dt = 100.0
+
+    _MSLP = 1e5
+    _grav = 9.80616
+    _R_m = 287.0024093890231
+    γ = 1.4
+    _C_p = _R_m * γ / (γ - 1)
+    _C_v = _R_m / (γ - 1)
+
+
+    P(ρθ) = _MSLP^(-_R_m/_C_v)*_R_m^(_C_p/_C_v)*(ρθ)^(_C_p/_C_v)
+    dP(ρθ) = _C_p/_C_v * _MSLP^(-_R_m/_C_v)*_R_m^(_C_p/_C_v)*(ρθ)^(_C_p/_C_v - 1)
+
+
+    # Initialization
+    ρ = zeros(Float64, N)
+    ρθ = zeros(Float64, N)
+    w = zeros(Float64, N+1)
+    Δz = zeros(N+1) .+ L/N
+    for i = 1:N
+        z = (i-0.5)*L/N
+        Tvᵢ, pᵢ, ρᵢ = DecayingTemperatureProfile(z; T_virt_surf = 300.0, T_min_ref = 300.0, _R_m = _R_m, _grav = _grav, _MSLP = _MSLP)
+        ρ[i]  = ρᵢ
+        ρθ[i] = ρᵢ*Tvᵢ*(_MSLP/pᵢ)^(_R_m/_C_p)
+    end
+    ss = Single_Stack(L, N, 0.0, Δz, copy(ρ), copy(w), copy(ρθ),copy(ρ), copy(w), copy(ρθ), _grav, γ, _R_m,  P, dP)
+    u0 = zeros(3*N+1)
+    combine!(ρ, w, ρθ, u0)
+
+    # ARS solve
+    single_column_prob_wfact_split = ODEProblem(
+        SplitFunction(
+            ODEFunction(
+                spatial_residual!;
+                jac_prototype=zeros(Float64,3N+1,3N+1),
+                Wfact = single_column_Wfact!,
+            ),
+            ODEFunction((du, u, p, t) -> (du .= 0.0)),
+        ),
+        u0,(0.0, N_iter*dt), ss)
+
+    algorithm = ARS343(
+        NewtonsMethod(; linsolve = linsolve_direct, max_iters = 1),
+    ) 
```

Comment:
@Zhengyu-Huang, to address Tapio's comment, we can define `algorithms` (plural) to be a tuple, as:
```suggestion
    algorithms = (ARS343(
            NewtonsMethod(; linsolve = linsolve_direct, max_iters = 1),
        ),
    )
```
and then pass this in a loop to the solve, as:

```suggestion
    for algo in algorithms
        u = solve(single_column_prob_wfact_split , algo; dt=dt)
    end
```
This should work (just double check the syntax). Let me know if you need help with that.